src/bin/rewrite.rs

```diff
@@ -136,10 +113,10 @@ fn main() {
 
             // Rewrite back the lambda and optionally rewrite back the DC invention format.
             for (i, pretty_program) in rewritten.split_programs().iter().enumerate() {
-                let task_name = program_id_to_task_name[&i].clone();
+                let task_name = input.tasks[i].clone();
                 let mut pretty_program = pretty_program.to_string();
                 if args.dreamcoder_output {
-                    for (name, dc_translation) in dreamcoder_translation.iter() {
+                    for (name, dc_translation) in dreamcoder_translation.iter().rev() {
```

Comment:
Why do we need to reverse the iterator here / why did we not need to do it before?

Reply:
Yeah, I think that in truth we don't actually need to reverse the iterator, since we're now turning fn_X functions in #xxxx functions, not the other way around. This actually shouldn't make any difference.